internal/backends/postgresql/metadata/registry.go

```diff
@@ -73,6 +115,108 @@ func (r *Registry) Close() {
 	r.p.Close()
 }
 
+// initDBs returns a list of database names using schema information.
+// It fetches existing schema (excluding ones reserved for postgresql),
+// then finds and returns schema that contains ferretdb metadata table.
+func (r *Registry) initDBs(ctx context.Context, p *pgxpool.Pool) ([]string, error) {
+	// schema names with pg_ prefix are reserved for postgresql hence excluded,
+	// a collection cannot be created in a database with pg_ prefix
+	q := `
+		SELECT schema_name
+		FROM information_schema.schemata
+		WHERE schema_name NOT LIKE 'pg_%'`
+
+	rows, err := p.Query(ctx, q)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	defer rows.Close()
+
+	if err = rows.Err(); err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	var dbNames []string
+
+	for rows.Next() {
+		var dbName string
+		if err = rows.Scan(&dbName); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		// schema created by postgresql (such as `public`) can be used as
+		// a ferretdb database, but if it does not contain ferretdb metadata table,
+		// it is not used by ferretdb
+		q := `
+			SELECT EXISTS (
+				SELECT 1
+				FROM information_schema.columns
+				WHERE table_schema = $1 AND table_name = $2
+				)`
+
+		var exists bool
+		if err = p.QueryRow(ctx, q, dbName, metadataTableName).Scan(&exists); err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+
+		if exists {
+			dbNames = append(dbNames, dbName)
+		}
+	}
+
+	return dbNames, nil
+}
+
+// initCollections loads collections metadata from the database during initialization.
+func (r *Registry) initCollections(ctx context.Context, dbName string, p *pgxpool.Pool) error {
+	defer observability.FuncCall(ctx)()
+
+	r.rw.Lock()
```

Comment:
Why do we need a lock here? I see that for SQLite we don't have a lock in a similar place.